Index: a/themes/openSUSE/src/dia_install.inc
===================================================================
--- a.orig/themes/openSUSE/src/dia_install.inc	2009-11-25 13:33:37.000000000 +0000
+++ a/themes/openSUSE/src/dia_install.inc	2009-11-25 13:33:49.000000000 +0000
@@ -49,13 +49,8 @@
   xmenu .xm_list [
     is_dvd { "DVD" } { "CD-ROM" } ifelse
     /txt_harddisk
-    "\x09SLP"
-    "FTP"
-    "HTTP"
     "NFS"
-    "SMB / CIFS"
     /install.submenu.net
-    /txt_http_proxy
   ] put
 
   xmenu .xm_title /txt_install_source put
