Index: gfxboot-themes-sidux/themes/openSUSE/src/gfxboot.cfg
===================================================================
--- gfxboot-themes-sidux.orig/themes/openSUSE/src/gfxboot.cfg	2009-11-25 14:53:38.000000000 +0000
+++ gfxboot-themes-sidux/themes/openSUSE/src/gfxboot.cfg	2009-11-25 14:55:39.000000000 +0000
@@ -100,7 +100,7 @@
 
 [install]
 ; show welcome animation
-welcome=1
+welcome=0
 ; beep when menu is ready
 beep=1
 ; main menu item shown only if iso is tagged for media check
