adapt color theme for aptosid.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/themes/openSUSE/src/gfxboot.cfg
+++ b/themes/openSUSE/src/gfxboot.cfg
@@ -48,7 +48,7 @@ mainmenu.selected.fg=0x000000
 bootopt.label.fg=0xffffff
 bootopt.text.fg=0xffffff
 ; progress bar color
-progress.bar.color=0x295202
+progress.bar.color=0xad1318
 ; menu colors
 menu.normal.bg=0xd8deed
 menu.normal.fg=0x000000
@@ -58,7 +58,7 @@ menu.selected.fg=0xffffff
 ; window colors
 window.bg=0xd8deed
 window.fg=0x000000
-window.title.bg=0x217b05
+window.title.bg=0x7d1117
 window.title.fg=0xffffff
 window.title.error.bg=0xc00000
 ; help system colors
